Mirantis/cri-dockerd
# Add a GHA for building a release

Fixes #206

## Proposed Changes
Add a release workflow that calls multiple other jobs to create and publish a release. The workflow will be triggered by creating a pre-release on the project's github page.  This will build and test all release artifacts. If everything is successful, it will publish the artifacts to the release page and push the latest doc changes. Afterwards, the release can be manually changed to ready for production.

- Builds and tests will all run in parallel. After which, publishing the docs and artifacts will be done in parallel
- PRs will still build linux, mac, and windows but no packages
- The artifact version follows what it placed on the github release page without a 'v'

![image](https://github.com/Mirantis/cri-dockerd/assets/5108029/313a22f0-3f6e-45df-9aff-e4da17288844)

## Testing
I tested this by running it on my personal fork of the repo. 

1. Create a pre-release
2. Watch for everything in the CI to pass. The hugo page was not setup on my fork and so this failed during publishing as expected
3. Verify that the correct release files were uploaded to the release and followed the current naming conventioin
